src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -2011,6 +2011,18 @@ int sockinfo_tcp::bind(const sockaddr *__addr, socklen_t __addrlen)
 		return -1; //error
 	}
 
+	if (!m_bound.is_anyaddr() && !m_bound.is_local_loopback()) {
+		ip_address local_ip(m_bound.get_in_addr());
+
+		net_device_resources_t* p_nd_resources = get_nd_resources((const ip_address)local_ip);
+		if (p_nd_resources) {
+			/* Ring is available now but not ready to receive */
+			m_p_rx_ring = p_nd_resources->p_ring;
```

Comment:
issue#1:
My logic is following:
we do refcnt++ in attach_receiver() (see https://github.com/Mellanox/libvma/blob/experimental_256B/src/vma/sock/sockinfo.cpp#L459) and refcnt-- in detach_receiver() (see: https://github.com/Mellanox/libvma/blob/experimental_256B/src/vma/sock/sockinfo.cpp#L535)
Doing refcnt++ in bind() does not allow making detach flow.